base/src/iosMain/kotlin/utils/Decimal.kt

```diff
@@ -291,39 +196,22 @@ else
         )
     )
 
-/**
- * Converts a [Number] to a [Decimal]
- */
-actual fun Number.toDecimal() = Decimal(NSDecimalNumber(this.toString()))
-
-/**
- * Converts a String to a [Decimal]
- */
-actual fun String.toDecimal() = Decimal(NSDecimalNumber(this))
-
-/**
- * Gets the double value of a [Decimal]
- */
-actual fun Decimal.toDouble() = nsDecimal.toString().toDouble()
-
-/**
- * Gets the integer value of a [Decimal]
- */
-actual fun Decimal.toInt() = nsDecimal.intValue
+actual fun Number.toFiniteDecimal() = FiniteDecimal(NSDecimalNumber(this.toString()))
+actual fun String.toFiniteDecimal(): FiniteDecimal? {
```

Comment:
Can simplify with `when`:
```
 = when (val decimal = NSDecimalNumber(this)) {
    NSDecimalNumber.notANumber -> null
    else -> FIniteDecimal(decimal)
}
```
In common code, String.toDecimal() returns NaN in this case, here suddenly it's null.
I think we should be consistent in all places

Reply:
String.toDecimal() returns a Decimal, this returns a FiniteDecimal. So if it cannot be resolved to a FiniteDecimal, it should return null. This method is marked as internal on the common level for a reason. It should only be used within the module